### ISSUE TITLE
Unable to install 

### ISSUE THREAD

**Issue by BugDiver** [JacobFischer/netlinkwrapper#1]
I'm not able to install this module using `npm`.
## Stacktarce:

```
gyp WARN download NVM_NODEJS_ORG_MIRROR is deprecated and will be removed in node-gyp v4, please use NODEJS_ORG_MIRROR
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
  CXX(target) Release/obj.target/netlinksocket/netlinksocket.o
In file included from ../netlinksocket.cc:1:
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:275:10: warning: 'FatalException'
      is deprecated [-Wdeprecated-declarations]
  return FatalException(v8::Isolate::GetCurrent(), try_catch);
         ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:270:18: note: 'FatalException' has
      been explicitly marked deprecated here
NODE_EXTERN void FatalException(v8::Isolate* isolate,
                 ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:287:10: warning: 'Encode' is
      deprecated [-Wdeprecated-declarations]
  return Encode(v8::Isolate::GetCurrent(), buf, len, encoding);
         ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:278:34: note: 'Encode' has been
      explicitly marked deprecated here
NODE_EXTERN v8::Local<v8::Value> Encode(v8::Isolate* isolate,
                                 ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:298:10: warning: 'DecodeBytes' is
      deprecated [-Wdeprecated-declarations]
  return DecodeBytes(v8::Isolate::GetCurrent(), val, encoding);
         ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:291:21: note: 'DecodeBytes' has
      been explicitly marked deprecated here
NODE_EXTERN ssize_t DecodeBytes(v8::Isolate* isolate,
                    ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:312:10: warning: 'DecodeWrite' is
      deprecated [-Wdeprecated-declarations]
  return DecodeWrite(v8::Isolate::GetCurrent(), buf, buflen, val, encoding);
         ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:302:21: note: 'DecodeWrite' has
      been explicitly marked deprecated here
NODE_EXTERN ssize_t DecodeWrite(v8::Isolate* isolate,
                    ^
4 warnings generated.
  CXX(target) Release/obj.target/netlinksocket/netlinkwrapper.o
In file included from ../netlinkwrapper.cc:1:
In file included from ../netlinkwrapper.h:4:
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:275:10: warning: 'FatalException'
      is deprecated [-Wdeprecated-declarations]
  return FatalException(v8::Isolate::GetCurrent(), try_catch);
         ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:270:18: note: 'FatalException' has
      been explicitly marked deprecated here
NODE_EXTERN void FatalException(v8::Isolate* isolate,
                 ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:287:10: warning: 'Encode' is
      deprecated [-Wdeprecated-declarations]
  return Encode(v8::Isolate::GetCurrent(), buf, len, encoding);
         ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:278:34: note: 'Encode' has been
      explicitly marked deprecated here
NODE_EXTERN v8::Local<v8::Value> Encode(v8::Isolate* isolate,
                                 ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:298:10: warning: 'DecodeBytes' is
      deprecated [-Wdeprecated-declarations]
  return DecodeBytes(v8::Isolate::GetCurrent(), val, encoding);
         ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:291:21: note: 'DecodeBytes' has
      been explicitly marked deprecated here
NODE_EXTERN ssize_t DecodeBytes(v8::Isolate* isolate,
                    ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:312:10: warning: 'DecodeWrite' is
      deprecated [-Wdeprecated-declarations]
  return DecodeWrite(v8::Isolate::GetCurrent(), buf, buflen, val, encoding);
         ^
/Users/vinaysh/.node-gyp/0.12.10/include/node/node.h:302:21: note: 'DecodeWrite' has
      been explicitly marked deprecated here
NODE_EXTERN ssize_t DecodeWrite(v8::Isolate* isolate,
                    ^
../netlinkwrapper.cc:14:25: error: use of undeclared identifier 'nullptr'
    if (this->socket != nullptr)
                        ^
../netlinkwrapper.cc:96:83: error: use of undeclared identifier 'e'
        isolate->ThrowException(Exception::TypeError(String::NewFromUtf8(isolate, e.what...
                                                                                  ^
../netlinkwrapper.cc:117:87: error: use of undeclared identifier 'e'
  ...isolate->ThrowException(Exception::TypeError(String::NewFromUtf8(isolate, e.what...
                                                                               ^
../netlinkwrapper.cc:136:87: error: use of undeclared identifier 'e'
  ...isolate->ThrowException(Exception::TypeError(String::NewFromUtf8(isolate, e.what...
                                                                               ^
../netlinkwrapper.cc:171:83: error: use of undeclared identifier 'e'
        isolate->ThrowException(Exception::TypeError(String::NewFromUtf8(isolate, e.what...
                                                                                  ^
../netlinkwrapper.cc:175:38: warning: comparison of integers of different signs: 'int'
      and 'size_t' (aka 'unsigned long') [-Wsign-compare]
    if(bufferRead > -1 && bufferRead <= bufferSize) // range check
                          ~~~~~~~~~~ ^  ~~~~~~~~~~
../netlinkwrapper.cc:206:83: error: use of undeclared identifier 'e'
        isolate->ThrowException(Exception::TypeError(String::NewFromUtf8(isolate, e.what...
                                                                                  ^
../netlinkwrapper.cc:224:83: error: use of undeclared identifier 'e'
        isolate->ThrowException(Exception::TypeError(String::NewFromUtf8(isolate, e.what...
                                                                                  ^
In file included from ../netlinkwrapper.cc:1:
../netlinkwrapper.h:28:16: warning: private field 'value_' is not used
      [-Wunused-private-field]
        double value_;
               ^
6 warnings and 7 errors generated.
make: *** [Release/obj.target/netlinksocket/netlinkwrapper.o] Error 1
gyp ERR! build error
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at ChildProcess.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1075:12)
gyp ERR! System Darwin 15.6.0
gyp ERR! command "node" "/usr/local/bin/node-gyp" "rebuild"
gyp ERR! cwd /Users/vinaysh/node_modules/netlinkwrapper
gyp ERR! node -v v0.12.10
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok
```

**Comment by JacobFischer** [comment]
Hey sorry I missed your issue for so long (forgot to star my own repo :P)

What OS are you on? I couldn't discern exactly based on the build output.

If this issue is too old and you don't care anymore I'll just close the issue.

**Comment by BugDiver** [comment]
MacBook Pro (Retina, 15-inch, Early 2013)
8 GB 1600 MHz DDR3

**Comment by JacobFischer** [comment]
Try the most recent commit, which is also the newly released version v1.0.0. I had never personally tried to compile it on OSX but thanks to @user404d he got it to work. Hopefully that fixes your issues

**Comment by BugDiver** [comment]
Thanks a lot to @user404d. Now I'm able to install it.
